packages/dds/merge-tree/src/client.ts

```diff
@@ -996,18 +997,46 @@ export class Client {
         }
     }
 
-    getContainingSegment<T extends ISegment>(pos: number, op?: ISequencedDocumentMessage) {
-        let seq: number;
-        let clientId: number;
+    private getRefSeqAndClientId(op?: ISequencedDocumentMessage) {
         if (op) {
-            clientId = this.getOrAddShortClientId(op.clientId);
-            seq = op.referenceSequenceNumber;
+            return { refSeq: op.referenceSequenceNumber, clientId: this.getOrAddShortClientId(op.clientId) };
         } else {
             const segWindow = this.mergeTree.getCollabWindow();
-            seq = segWindow.currentSeq;
-            clientId = segWindow.clientId;
+            return { refSeq: segWindow.currentSeq, clientId: segWindow.clientId };
         }
-        return this.mergeTree.getContainingSegment<T>(pos, seq, clientId);
+    }
+
+    getContainingSegment<T extends ISegment>(pos: number, op?: ISequencedDocumentMessage) {
+        const { refSeq, clientId } = this.getRefSeqAndClientId(op);
+        return this.mergeTree.getContainingSegment<T>(pos, refSeq, clientId);
+    }
+
+    createSlideOnRemoveReference(refType: ReferenceType, pos: number, pending: boolean, op?: ISequencedDocumentMessage):
```

Comment:
rather than a new an specific function like this, i'd rather just need the necessary logic be part of `createLocalReferencePosition` in #10193

Reply:
I chose a new method because:
1. I wanted the logic that handles sliding on create conflict in merge-tree not sequence. That requires passing a position instead of segment and offset, as well as the op parameter. As a result the method signature is very different from createLocalReferencePosition.
2. I wanted it to be clear that this method is paired with the ack method and easier to understand when that needs to be called.